### PR TITLE
Handle permissions in admin extension

### PR DESCRIPTION
**Pull request by core23**
When adding some routes with `configureRoutes`, there should be a way to add the corresponding access mappings.

### CODE REVIEW

File: Admin/AdminExtensionInterface.php [hunk 1]
```diff
@@ -123,6 +123,15 @@ public function alterObject(AdminInterface $admin, $object);
     public function getPersistentParameters(AdminInterface $admin);
 
     /**
+     * Return the controller access mapping.
+     *
+     * @param AdminInterface $admin
+     *
+     * @return array
+     */
+    public function getAccessMapping(AdminInterface $admin);
```
Comment (Soullivaneuh) [inline Admin/AdminExtensionInterface.php]: Adding a method to an interface is BC break.

**Comment by core23** [comment]
Added BC info to changelog file

**Comment by rande** [comment]
+1

**Comment by Soullivaneuh** [comment]
> Added BC info to changelog file

This is not the good way. The BC break need to be handled properly with deprecation notice.

If it's not done, this PR will be merged only for next major (3.0). That could take a quite long time.

**Comment by rande** [comment]
@Soullivaneuh I think we can go for a BC break for the 2.4-develop branch. But it is my 2c.

**Comment by Soullivaneuh** [comment]
> I think we can go for a BC break for the 2.4-develop branch

I'm sorry but I totally :-1: for this.

No BC break should be introduced in any minor version according to semver.

BC break is a nightmare that every developer will have while running a simple composer update if we don't follow this rule.

For this case. This could be added outside the interface with a deprecation warning to be added later.

**Comment by core23** [comment]
Removed the BC break, so we could merge this in the current branch.

**Comment by OskarStark** [comment]
Thank you @core23 

can you provide a PR with the changes for the next major and reference it here?

Thank you :beers:

**Comment by core23** [comment]
@OskarStark please remove the 3.0 milestone flag for this

**Comment by OskarStark** [comment]
:+1:

**Pull request merged**